### DISCUSSION TITLE
How do I create a debug 2D grid overlay widget?

### DISCUSSION THREAD

**Discussion by vini-fda** [emilk/egui#2899] [Q&A]
I'm currently developing a simple 2D simulation with graphics using [macroquad](https://github.com/not-fl3/macroquad). In my debug view, I want to precisely know the world coordinates pointed at by the mouse, and also want to have a grid to better estimate distances etc.

Currently, I'm using egui's `Plot` with transparent background as my overlay, but I'm thinking about creating my own widget for this, because there are many options that aren't present in the public interface of the `Plot` widget, such as changing the gridline colors. Also, evidently, the Plot widget wasn't even made with this application in mind. Any pointers on how to create my own implementation without "reinventing the wheel"?

![image](https://user-images.githubusercontent.com/20410205/231844059-5645aa25-a7a5-47c2-94c3-e9ccea66330d.png)

**Comment by jptrzy** [comment]
Check the original source of the egui plot - it is quite interesting piece of code.
